src/runtime_src/core/common/thread.cpp

```diff
@@ -140,8 +140,47 @@ set_thread_policy(std::thread&)
 }
 
 static void
-set_cpu_affinity(std::thread&)
+set_cpu_affinity(std::thread& thread)
 {
+  static bool initialized = false;
+  static DWORD_PTR affinityMask = 0;
+  static bool all = false;
```

Comment:
Let's fix this here and for Linux as well.  Remove this `all` variable and replace all occurrences of `all = true;` with `return;`.  Drop the outer `else` and reduce indentation with early-exit.